x-pack/plugins/security_solution/common/api/entity_analytics/asset_criticality/common.gen.ts

```diff
@@ -53,8 +53,37 @@ export const CreateAssetCriticalityRecord = AssetCriticalityRecordIdParts.merge(
   })
 );
 
+export type AssetCriticalityRecordEcsParts = z.infer<typeof AssetCriticalityRecordEcsParts>;
```

Comment:
Nit: 
I assume we need "Ecs" in the name to differentiate from the current types?

Reply:
Yes. The `AssetCriticalityRecord` was already broken down into two types: `AssetCriticalityRecordIdParts`, and `CreateAssetCriticalityRecord`

So I created a new type with ECS fields and updated `AssetCriticalityRecord` to be composed of `CreateAssetCriticalityRecord` + `AssetCriticalityRecordEcsParts` + '@timestamp' field.